examples/keras_io/vision/image_classification_using_global_context_vision_transformer.py

```diff
@@ -0,0 +1,1182 @@
+    1. `Pooling:` It converts $N \times 2D$ features to $N \times 1D$ features.
+    2. `Classifier:` It processes $N \times 1D$ features to make a decision about class.
+
+I've annotated the architecture figure to make it easier to digest,
+<img src="https://raw.githubusercontent.com/awsaf49/gcvit-tf/main/image/arch_annot.png">
+"""
+
+"""
+### Unit Blocks
+
+> **Note:** This blocks are used to build other modules throughout the paper. Most of the
+blocks are either borrowed from other work or modified version old work.
+
+1. `SE`: **Squeeze-Excitation (SE)** aka **Bottleneck** module acts sd kind of **channel
+attention**. It consits of **AvgPooling**, **Dense/FullyConnected (FC)/Linear** ,
+**GELU** and **Sigmoid** module.
+<img src="https://raw.githubusercontent.com/awsaf49/gcvit-tf/main/image/se_annot.png"
+width=400>
+
+2. `Fused-MBConv:` This is similar to the one used in **EfficientNetV2**. It uses
+**Depthwise-Conv**, **GELU**, **SE**, **Conv**, to extract feature with a resiudal
+connection. Note that, no new module is declared for this one, we simply applied
+corresponding modules directly.
+<img src="https://raw.githubusercontent.com/awsaf49/gcvit-tf/main/image/fmb_annot.png"
+width=350>
+
+3. `ReduceSize`: It is a **CNN** based **downsample** module which abvobe mentioned
+`Fused-MBConv` module to extract feature, **Strided Conv** to simultaneously reduce
+spatial dimension and increse channelwise dimention of the features and finally
+**LayerNormalization** module to normalize features. In the paper/figure this module is
+referred as **downsample** module. I think it is mention worthy that **SwniTransformer**
+used `PatchMerging` module instead of `ReduceSize` to reduce the spatial dimention and
+increase channelwise dimension which uses **fully-connected/dense/linear** module.
+According to the **GCViT** paper, one of the purposes of using `ReduceSize` is to add
+inductive bias through **CNN** module.
+<img src="https://raw.githubusercontent.com/awsaf49/gcvit-tf/main/image/down_annot.png"
+width=300>
+
+4. `MLP:` This is our very own **Multi Layer Perceptron** module. This a
+feed-forward/fully-connected/linear module which simply projects input to an arbitary
+dimension.
+"""
+
+
+class SE(keras.layers.Layer):
+    """Squeeze and excitation block.
+
+    Args:
+        output_dim: output features dimension, if `None` use same dim as input.
+        expansion: expansion ratio.
+    """
+
+    def __init__(self, output_dim=None, expansion=0.25, **kwargs):
+        super().__init__(**kwargs)
+        self.expansion = expansion
+        self.output_dim = output_dim
+
+    def build(self, input_shape):
+        inp = input_shape[-1]
+        self.output_dim = self.output_dim or inp
+        self.avg_pool = keras.layers.GlobalAvgPool2D(keepdims=True, name="avg_pool")
+        self.fc = [
+            keras.layers.Dense(int(inp * self.expansion), use_bias=False, name="fc_0"),
```

Comment:
You'll save a bunch of characters if you import `layers` and access layers from there